core/src/main/java/org/pentaho/di/core/database/Database.java

```diff
@@ -3677,7 +3677,11 @@ public RowMetaInterface getReturnRowMeta() {
   }
 
   public String[] getTablenames( String schemanamein, boolean includeSchema ) throws KettleDatabaseException {
-    Map<String, Collection<String>> tableMap = getTableMap( schemanamein );
+    return getTablenames( null, includeSchema, null );
```

Comment:
shouldn't this be getTablenames( schemanamein, includeSchema, null ); instead of null in the first parameter?